Signal/src/ViewControllers/FingerprintViewController.m

```diff
@@ -63,59 +80,234 @@ - (void)configureWithRecipientId:(NSString *)recipientId
+                                                                     target:self
+                                                                     action:@selector(didTapShareButton)];
     self.navigationItem.rightBarButtonItem = self.shareButton;
-    [self.modalNavigationBar pushNavigationItem:self.navigationItem animated:NO];
-
-    // HACK for transparent navigation bar.
-    [self.modalNavigationBar setBackgroundImage:[UIImage new] forBarMetrics:UIBarMetricsDefault];
-    self.modalNavigationBar.shadowImage = [UIImage new];
-    self.modalNavigationBar.translucent = YES;
-
-    // HACK to get full width preview layer
-    CGRect oldFrame = self.qrScanningView.frame;
-    CGRect newFrame = CGRectMake(oldFrame.origin.x,
-        oldFrame.origin.y,
-        self.view.frame.size.width,
-        self.view.frame.size.height / 2.0f - oldFrame.origin.y);
-    self.qrScanningView.frame = newFrame;
-    // END HACK to get full width preview layer
 
-    self.title = NSLocalizedString(@"PRIVACY_VERIFICATION_TITLE", @"Navbar title");
-    NSString *instructionsFormat = NSLocalizedString(@"PRIVACY_VERIFICATION_INSTRUCTIONS",
-        @"Paragraph(s) shown alongside the safety number when verifying privacy with {{contact name}}");
-    self.instructionsLabel.text = [NSString stringWithFormat:instructionsFormat, self.contactName];
+    [self createViews];
+}
 
-    NSString *scanTitle = NSLocalizedString(@"SCAN_CODE_ACTION",
-        @"Button label presented with camera icon while verifying privacy credentials. Shows the camera interface.");
-    [self.scanButton setTitle:scanTitle forState:UIControlStateNormal];
-    self.scanningInstructions.text
+- (void)createViews
+{
+    UIColor *darkGrey = [UIColor colorWithRGBHex:0x404040];
+
+    self.view.backgroundColor = [UIColor whiteColor];
+
+    UIView *referenceView = [UIView new];
+    self.referenceView = referenceView;
+    [self.view addSubview:referenceView];
+    [referenceView autoPinWidthToSuperview];
+    [referenceView autoPinToTopLayoutGuideOfViewController:self withInset:0];
+    [referenceView autoPinToBottomLayoutGuideOfViewController:self withInset:0];
+
+    UIView *mainView = [UIView new];
+    self.mainView = mainView;
+    mainView.backgroundColor = [UIColor whiteColor];
+    [self.view addSubview:mainView];
+    [mainView autoPinWidthToSuperview];
+    [[NSLayoutConstraint constraintWithItem:mainView
+                                  attribute:NSLayoutAttributeHeight
+                                  relatedBy:NSLayoutRelationEqual
+                                     toItem:referenceView
+                                  attribute:NSLayoutAttributeHeight
+                                 multiplier:1.0
+                                   constant:0.f] autoInstall];
+    [mainView
+        addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(mainViewTapped:)]];
```

Comment:
I've made more things tappable.  Before, if you tapped the safety number text, it presented the sharing UI.  I now do the same if you tap on the QR code.  I also made the hot area for the "scan code" button larger. Lastly, I made the entire body tappable - it dismisses the camera/scanning views if visible.